hera_cal/redcal.py

```diff
@@ -269,6 +337,127 @@ def multiply_by_gains(target, gains, target_type='vis'):
 
     return _apply_gains(target, gains, (lambda x, y: x * y), target_type)
 
+class OmnicalSolver(linsolve.LinProductSolver):
+    def __init__(self, data, sol0, wgts={}, gain=.3, **kwargs):
+        """Set up a nonlinear system of equations of the form g_i * g_j.conj() * V_mdl = V_ij
+        to linearize via the Omnical algorithm from Liu et al. 2010.
+
+        Args:
+            data: Dictionary that maps nonlinear product equations, written as valid python-interpetable 
+                strings that include the variables in question, to (complex) numbers or numpy arrarys. 
+                Variables with trailing underscores '_' are interpreted as complex conjugates (e.g. x*y_ 
+                parses as x * y.conj()).
+            sol0: Dictionary mapping all variables (as keyword strings) to their starting guess values.
+                This is the point that is Taylor expanded around, so it must be relatively close to the
+                true chi^2 minimizing solution. In the same format as that produced by 
+                linsolve.LogProductSolver.solve() or linsolve.LinProductSolver.solve().
+            wgts: Dictionary that maps equation strings from data to real weights to apply to each 
+                equation. Weights are treated as 1/sigma^2. All equations in the data must have a weight 
+                if wgts is not the default, {}, which means all 1.0s.
+            gain: The fractional step made toward the new solution each iteration.
+            **kwargs: keyword arguments of constants (python variables in keys of data that 
+                are not to be solved for)
+
+        Returns:
```

Comment:
I generally skip this when there are no returns, but that's your call.

Reply:
Done.